src/kafka-util/src/bin/kgen.rs

```diff
@@ -565,6 +573,29 @@ async fn main() -> anyhow::Result<()> {
         _ => unreachable!(),
     };
 
+    let mut key_gen = match matches.value_of("keys").unwrap() {
+        "avro" => {
+            let key_schema = matches.value_of("avro-key-schema").unwrap();
+            let ccsr =
+                ccsr::ClientConfig::new(Url::parse("http://localhost:8081").unwrap()).build();
+            let key_schema_id = ccsr
+                .publish_schema(&format!("{}-key", topic), key_schema)
+                .await?;
+            _key_schema_holder = Some(Schema::parse_str(key_schema)?);
+            let key_schema = _key_schema_holder.as_ref().unwrap();
+            let annotations = matches.value_of("avro-key-distribution").unwrap();
+            let annotations: serde_json::Value = serde_json::from_str(annotations)?;
+            let generator = RandomAvroGenerator::new(key_schema, &annotations);
+
+            Some(ValueGenerator::RandomAvro {
+                inner: generator,
+                schema: key_schema,
+                schema_id: key_schema_id,
+            })
+        }
+        _ => None,
```

Comment:
Perhaps consider printing an error message if either of `--avro-key-schema` or `--avro-key-definition` are defined but the `keys` type is not `avro`?

Reply:
I did something that accomplishes this in a roundabout way -- now `avro-key-schema` and `avro-key-definition` can no longer be defined alongside `key-min` and `key-max` which means the two `avro-key-*` args can't be used while `keys=random` which i think is good enough for now.